.github/workflows/docker.yml

```diff
@@ -3,12 +3,37 @@ name: Docker
 on:
   push:
     branches: [ main ]
-  pull_request: {}
+  pull_request:
+    branches: [ main ]
+  release:
+    types: [ published ]
 
 jobs:
   docker:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - name: Build Image
+
+    - uses: docker/metadata-action@v5
+      id: meta
+      with:
+        images: |
+          icinga/icinga-notifications
```

Comment:
Can you please summarize which tags this will push by default?

@lippserd I wonder why you [went with `type=edge` in Icinga for Kubernetes](https://github.com/Icinga/icinga-kubernetes/blob/854c171a436a6a4d951c618f9f44a2067fd8587d/.github/workflows/docker-image.yml#L20-L21) (I also wonder why it exists in the first place in metadata-action, it seems to do nothing special other than renaming the tag for the default branch to edge?)

Reply:
I went ahead playing with this in a private repo myself. Results:

### default

```yaml
# tags: (not set)
```

#### push to main

```
icinga/icinga-notifications:main
```

#### release published

```
icinga/icinga-notifications:v0.1.0
icinga/icinga-notifications:latest
```

**Note:** Release tag has `v` prefix in contrast to our other images (Icinga 2, Icinga Web, Icinga DB)

### edge + semver

```yaml
tags: |
  type=edge
  type=semver,pattern={{version}}
  type=semver,pattern={{major}}.{{minor}}
```

#### push to main

```
icinga/icinga-notifications:edge
```

#### release published

```
icinga/icinga-notifications:0.2.0
icinga/icinga-notifications:0.2
icinga/icinga-notifications:latest
```

### ref + semver

```yaml
tags: |
  type=ref,event=branch,enable={{is_default_branch}}
  type=semver,pattern={{version}}
  type=semver,pattern={{major}}.{{minor}}
```

#### push to main

```
icinga/icinga-notifications:main
```

#### release published

```
icinga/icinga-notifications:0.3.0
icinga/icinga-notifications:0.3
icinga/icinga-notifications:latest
```

### Summary

I'd suggest to go with the last variant (ref + semver), unless we decide to go for `:edge` as the snapshot/nightly/bleeding edge tag for all our images.

Reply:
I decided to use `edge` to indicate snapshot builds. At least that's what I've found is more commonly used than `snapshot` for example. At the time of the release of version 0.1.0, I didn't have the right rules configured, so an image with tag 0.1.0/latest was never uploaded and I didn't push one either, as I thought we'd release 0.2.0 faster.

Reply:
> unless we decide to go for `:edge` as the snapshot/nightly/bleeding edge tag for all our images

Sounds like a good idea to me.

Reply:
> I didn't have the right rules configured, so an image with tag 0.1.0/latest was never uploaded and I didn't push one either, as I thought we'd release 0.2.0 faster.

FYI: I think the [current version of the Icinga for Kubernetes](https://github.com/Icinga/icinga-kubernetes/blob/main/.github/workflows/docker-image.yml) won't upload for a tag/release as setting `type=edge` should disable all other rules that would by default set version tags.

Reply:
Settings all version tags properly will be another task. Keep in mind that this becomes extra tricky when you release versions out of order, i.e. release 2.14.0 and afterwards, you want to release a bug fix as 2.13.5. In Icinga 2, we currently have [this beautiful script](https://github.com/Icinga/docker-icinga2/blob/master/mktags.bash) for that.

For this project (and probably Icinga for Kubernetes as well), this won't be an issue until at least version 1.1.0. I think the general plan here is to use this here a bit as a playground for how to build and push the Docker images, next, that should be adapted for the existing images, where that issue should then be solved.

Reply:
> Settings all version tags properly will be another task. Keep in mind that this becomes extra tricky when you release versions out of order, i.e. release 2.14.0 and afterwards, you want to release a bug fix as 2.13.5. In Icinga 2, we currently have [this beautiful script](https://github.com/Icinga/docker-icinga2/blob/master/mktags.bash) for that.

Valid point, I have only thought about continuous release numbers.

> For this project (and probably Icinga for Kubernetes as well), this won't be an issue until at least version 1.1.0. I think the general plan here is to use this here a bit as a playground for how to build and push the Docker images, next, that should be adapted for the existing images, where that issue should then be solved.

Then I would not include it, or at least not at the moment.